File: sakuralearn-backend/src/main/java/com/sakuralearn/sakuralearn_backend/service/AuthService.java
package com.sakuralearn.sakuralearn_backend.service;

import com.sakuralearn.sakuralearn_backend.dto.request.LoginRequest;
import com.sakuralearn.sakuralearn_backend.dto.request.RegisterRequest;
import com.sakuralearn.sakuralearn_backend.dto.request.RefreshTokenRequest;
import com.sakuralearn.sakuralearn_backend.dto.response.AuthResponse;
import com.sakuralearn.sakuralearn_backend.dto.response.TokenRefreshResponse;

public interface AuthService {
    AuthResponse authenticateUser(LoginRequest loginRequest);
    void registerUser(RegisterRequest signUpRequest);
    TokenRefreshResponse refreshToken(RefreshTokenRequest request);
    boolean verifyEmail(String token);
    void logout(java.util.UUID userId);
    void forgotPassword(String email);
    void resetPassword(String token, String newPassword);
}

File: sakuralearn-frontend/src/services/authService.js
import API from './api'

export const authService = {
  login: (data) => API.post('/auth/login', data),
  register: (data) => API.post('/auth/register', data),
  logout: () => API.post('/auth/logout'),
  forgotPassword: (email) => API.post(`/auth/forgot-password?email=${email}`),
  resetPassword: (token, newPassword) => API.post(`/auth/reset-password?token=${token}&newPassword=${newPassword}`),
  refreshToken: (refreshToken) => API.post('/auth/refresh', { refreshToken }),
}
